### PR TITLE
Misc cleanup to flow code/tests pulled from other patches.

### PR DESCRIPTION
**Pull request by benvanik**

### CODE REVIEW

**Review by ScottTodd — APPROVED** [review]
Thanks for splitting these changes out into their own PR.

File: compiler/src/iree/compiler/Dialect/Flow/Conversion/TensorToFlow/Patterns.cpp [hunk 1]
```diff
@@ -20,6 +20,29 @@ namespace mlir::iree_compiler::IREE::Flow {
 
 namespace {
 
+/// Converts linalg.fill ops into flow.tensor.splat ops.
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Dialect/Flow/Conversion/TensorToFlow/Patterns.cpp:23]: Wait, we should already have this?

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]

File: compiler/src/iree/compiler/Dialect/Flow/Conversion/TensorToFlow/Patterns.cpp (hunk 1 above)
Comment (ScottTodd) [reply compiler/src/iree/compiler/Dialect/Flow/Conversion/TensorToFlow/Patterns.cpp:23]: It was moved from lower in the file

Comment (MaheshRavishankar) [reply compiler/src/iree/compiler/Dialect/Flow/Conversion/TensorToFlow/Patterns.cpp:23]: Oh NM... looks like you moved this.

File: compiler/src/iree/compiler/Dialect/Flow/Conversion/TensorToFlow/Patterns.cpp [hunk 2]
```diff
@@ -20,6 +20,29 @@ namespace mlir::iree_compiler::IREE::Flow {
...
+///
+/// This is expected to improve performance because we can use DMA
+/// functionalities for the fill, instead of dispatching kernels.
+struct ConvertLinalgFillPattern final
+    : public OpRewritePattern<linalg::FillOp> {
+  using OpRewritePattern::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(linalg::FillOp fillOp,
+                                PatternRewriter &rewriter) const override {
+    if (fillOp->getParentOfType<IREE::Flow::DispatchWorkgroupsOp>()) {
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Dialect/Flow/Conversion/TensorToFlow/Patterns.cpp:33]: I have been using this https://github.com/MaheshRavishankar/iree/blob/59f93aafc6a9465f7f3234dcdad76c8812d2f898/compiler/src/iree/compiler/Dialect/Flow/Transforms/RegionOpUtils.cpp#L122 . But this is fine.

**Review by MaheshRavishankar — APPROVED** [review]

File: compiler/src/iree/compiler/Dialect/Flow/Conversion/TensorToFlow/Patterns.cpp (hunk 1 above)
Comment (benvanik) [reply compiler/src/iree/compiler/Dialect/Flow/Conversion/TensorToFlow/Patterns.cpp:23]: it was out of order - this just moved it up

**Comment by benvanik** [comment]
test failure is a tsan flake

**Pull request merged**